src/java/org/apache/cassandra/index/sai/plan/StorageAttachedIndexSearcher.java

```diff
@@ -122,7 +122,10 @@ public PartitionIterator filterReplicaFilteringProtection(PartitionIterator full
     public UnfilteredPartitionIterator search(ReadExecutionController executionController) throws RequestTimeoutException
     {
         FilterTree filterTree = analyzeFilter();
-        if (command.isTopK())
+        Plan plan = controller.buildPlan();
+
+        // Can't check for `command.isTopK()` because the planner could optimize sorting out
```

Comment:
Do you have an example where we would optimize sorting out?

Reply:
`SELECT * FROM tab WHERE x = ? ORDER BY x`